### PR TITLE
[Enhancement] Remove db lock when planning olap table query

### LINKED ISSUE #15627: db lock take a long time when query or load
## Enhancement

1. if optimizer has some bugs,  query plan may need a long time when optimizer hold the db lock
2. When load  transaction commit， it also need a db lock and may take a long time db lock.

In fact，the db lock is too coarse，and we need fine-grained lock even no lock.

As a first step，I will make the query process don't rely on db lock

### PR DESCRIPTION
**Pull request by kangkaisen**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes https://github.com/StarRocks/starrocks/issues/15627

## Problem Summary(Required) ：

1. Remove the db lock when planning olap table query
2. deep copy the olap table meta when query analyze
3. when query optimizer, use the copied olap table meta all the time.

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 27 / 33 (81.82%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Table.java|0|1|00.00%|[561]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|2|00.00%|[394, 396]|
|:large_blue_circle:|com/starrocks/sql/StatementPlanner.java|23|26|88.46%|[67, 98, 99]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/tree/AddDecodeNodeForDictStringRule.java|1|1|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/QueryAnalyzer.java [hunk 1]
```diff
@@ -276,7 +277,13 @@ private Relation resolveTableRef(Relation relation, Scope scope, Set<TableName>
                     }
 
                     if (table.isSupported()) {
-                        tableRelation.setTable(table);
+                        if (table.isOlapTable()) {
+                            // Copying the olap table meta to avoid the lock when query
+                            Table copied = DeepCopy.copyWithGson(table, OlapTable.class);
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/QueryAnalyzer.java:282]: If you failed multi times to replan, I think you need to degeneration to lock to get rid of high concurrency load.

Comment (Seaven) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/QueryAnalyzer.java:282]: Why only copy olapTable？may use other type table direct without lock?

**Review by Seaven — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 27 / 27 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/MvRewritePreprocessor.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/QueryAnalyzer.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzerUtils.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/StatementPlanner.java|12|12|100.00%|[]|

**Review by satanson — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/sql/StatementPlanner.java [hunk 2]
```diff
@@ -58,21 +58,35 @@ public static ExecPlan plan(StatementBase stmt, ConnectContext session, boolean
...
             dbLocks = dbs;
-        }
-        try {
-            lock(dbLocks);
-            try (PlannerProfile.ScopedTimer ignored = PlannerProfile.getScopedTimer("Analyzer")) {
-                Analyzer.analyze(stmt, session);
+            try {
+                lock(dbLocks);
+                try (PlannerProfile.ScopedTimer ignored = PlannerProfile.getScopedTimer("Analyzer")) {
+                    Analyzer.analyze(stmt, session);
```
Comment (ZiheLiu) [inline fe/fe-core/src/main/java/com/starrocks/sql/StatementPlanner.java:68]: When the arguement `lockDb` is false, should we also analyze `stmt`?

**Pull request merged**